app/src/main/java/com/example/coronadiagnosticapp/ui/activities/OxymeterActivity.java

```diff
@@ -238,17 +243,22 @@ public double calculateSPO2(ArrayList<Double> red, ArrayList<Double> blue) {
             return spo2;
         }
 
-        public ArrayList<Double> calculateMovingAverage(ArrayList<Double> list) {
-            //Initialize an object that calculates the rolling average of last 15 samples
-            SMA calc_mov_avg = new SMA(15);
+        public ArrayList<Double> calculateMovingRedWindowAverage(ArrayList<Double> list) {
             ArrayList<Double> MovAvgRed = new ArrayList<Double>();
-            double avg_hr = (sumDouble(list)) / (list.size());
-            for (int i = 0; i < list.size(); i++) {
-                if (i < 15) {                                   //Assign the average red received to the first 15 samples
-                    calc_mov_avg.compute(avg_hr);               //Add the value to the moving average object
-                    MovAvgRed.add(i, avg_hr);                   //Add the value to the MobAvgRed list
-                } else {
-                    MovAvgRed.add(calc_mov_avg.compute(list.get(i)));
+            if (RedRollingAvg == null) {
+                //Initialize an object that calculates the rolling average of last 15 samples
+                RedRollingAvg = new SMA(15);
```

Comment:
Let's move the 15 to be a constant we define once and then use everywhere, other than that all looks good